src/SearchResults.js

```diff
@@ -0,0 +1,46 @@
+import React from "react";
+import moment from "moment";
+
+const SearchResults = props => {
+  return (
+    <table className="table">
+      <thead>
+        <tr>
+          {Object.keys(props.results[0]).map(function(key) {
+            return (
+              <th key={(-Math.random() * 10000000000).toString()} scope="col">
+                {key}
+              </th>
+            );
+          })}
+          <th scope="col">Number of nights</th>
+        </tr>
+      </thead>
+      <tbody>
+        {props.results.map(result => {
+          return (
+            <tr key={(Math.random() / 10000000000).toString()}>
+              {Object.keys(result).map(function(key) {
+                return (
+                  <td key={(Math.random() * 10000000000).toString()}>
+                    {result[key]}
+                  </td>
+                );
+              })}
+              <td>{NumberNights(result)}</td>
+            </tr>
+          );
+        })}
+      </tbody>
+    </table>
+  );
+};
+
+function NumberNights(result) {
+  let checkIn = moment(result.checkInDate);
+  let checkOut = moment(result.checkOutDate);
+
+  return checkOut.diff(checkIn, "days");
+}
+
```

Comment:
Good. One thing you can try to start doing is take note of when when you're creating variables that will never be altered or replaced (within that scope). In this case you set checkIn and checkOut once each and never need to change them again. This means you *could* use `const` rather than `let`. This doesn't change what the code does, but doing so can have the advantage of making it easier to spot if you are changing or replacing a value unintentionally later in your code as you'll get an error.